openfeature/openfeature.go

```diff
@@ -23,6 +23,10 @@ func SetProvider(provider FeatureProvider) error {
 	return api.setProvider(provider)
 }
 
+func SetProviderAndWait(provider FeatureProvider) error {
```

Comment:
+1 

I am fine with this method of naming. Please add a doc comment explaining the functionality. Also, please update `SetProvider` (L20-L24) doc comment mention that it is asynchronous (non-blocking)

Reply:
Added doc comments